src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -521,6 +519,7 @@ export default class QueryEditorUI extends Component<Props, State> {
               </EuiFlexGroup>
             </EuiFlexItem>
             <EuiFlexItem onClick={this.onClickInput} grow={true}>
+              <div ref={this.props.queryEditorRef} />
```

Comment:
nit: can this be renamed to like queryEditorHeaderRef? to avoid confusion. I'm fine with this name as well since we can think of this and the code editor the same items essentially.

Reply:
do we want to add a class and styling on this div as well?

two things we should consider adding is a max-height and overflow scroll to restrict how long this could get based what is being portal'd in